library/VTKExtensions/Rendering/vtkF3DDropZoneActor.cxx

```diff
@@ -0,0 +1,131 @@
+#include "vtkF3DDropZoneActor.h"
+
+#include <vtkObjectFactory.h>
+#include <vtkPolyData.h>
+#include <vtkPolyDataMapper2D.h>
+#include <vtkTextMapper.h>
+#include <vtkTextProperty.h>
+#include <vtkViewport.h>
+
+vtkStandardNewMacro(vtkF3DDropZoneActor);
+
+//------------------------------------------------------------------------------
+vtkF3DDropZoneActor::vtkF3DDropZoneActor()
+{
+  this->TextActor->SetMapper(this->TextMapper);
+  vtkTextProperty* tProp = this->TextMapper->GetTextProperty();
+  tProp->SetFontSize(25);
+  tProp->SetJustificationToCentered();
+  tProp->SetVerticalJustificationToCentered();
+
+  this->BorderMapper->SetInputData(this->BorderPolyData);
+  this->BorderActor->SetMapper(this->BorderMapper);
+}
+
+//------------------------------------------------------------------------------
+vtkF3DDropZoneActor::~vtkF3DDropZoneActor() = default;
+
+//------------------------------------------------------------------------------
+void vtkF3DDropZoneActor::ReleaseGraphicsResources(vtkWindow* win)
+{
+  this->TextActor->ReleaseGraphicsResources(win);
+  this->BorderActor->ReleaseGraphicsResources(win);
+}
+
+//------------------------------------------------------------------------------
+int vtkF3DDropZoneActor::RenderOverlay(vtkViewport* viewport)
+{
+  const int* vSize = viewport->GetSize();
+
+  // Simply position text actor in the middle
+  this->TextMapper->SetInput(this->DropText.c_str());
+  this->TextActor->SetPosition(vSize[0] / 2, vSize[1] / 2);
+  this->TextActor->RenderOverlay(viewport);
+
+  if (this->BuildBorderGeometry(viewport))
+  {
+    this->BorderActor->RenderOverlay(viewport);
+  }
+  return 1;
+}
+
+//------------------------------------------------------------------------------
+bool vtkF3DDropZoneActor::BuildBorderGeometry(vtkViewport* viewport)
+{
+  constexpr int tickWidth = 3;
+  constexpr int tickLength = 10;
+
+  const int* vSize = viewport->GetSize();
+  if (vSize[0] == this->ComputedBorderViewportSize[0] &&
+    vSize[1] == this->ComputedBorderViewportSize[1])
+  {
+    return true;
+  }
+
+  // padding is 10% of shortest side
+  const int padding = std::min(vSize[0], vSize[1]) * 0.1;
+  const int borderW = vSize[0] - 2 * padding;
+  const int borderH = vSize[1] - 2 * padding;
+
+  // number of ticks assuming spacing == tickLength
+  const int nTicksW = borderW / (tickLength * 2);
```

Comment:
Rounding up here makes a minor difference for small dimensions

casting to int:
![dash-cast](https://github.com/f3d-app/f3d/assets/489715/836d7b98-8436-4631-a038-96fcb0c960b7)

using `ceil`:
![dash-ceil](https://github.com/f3d-app/f3d/assets/489715/4f028c3b-8cdc-49ce-ad41-913dd08003ff)

Reply:
my mistake indeed